### PR TITLE
Filtering Alerts (TR-56)

### PR DESCRIPTION
**Pull request by njteh7**

### CODE REVIEW

File: client/src/Components/Alerts/AlertList.js [hunk 1]
```diff
@@ -86,14 +88,14 @@ const AlertList = () => {
             </div>
           </div>
         </div>
-        {list.map(alert => (
+        {list.filter(alert => alert.creator == creatorId).map(alert => (
```
Comment (ashwin319) [inline client/src/Components/Alerts/AlertList.js:91]: this approach works well; however, I find it more readable to move more complex logic out of the html template and into the controller.

I suggest making this line:
`{alertsByCreator.map(alert => (`

and define `alertsByCreator` above where you're fetching the creatorId

File: client/src/Components/Alerts/AlertList.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
 import React, { useState, useEffect } from 'react';
 import axios from 'axios';
-import { Redirect, useHistory} from 'react-router-dom';
+import { Redirect, useHistory } from 'react-router-dom';
 
 import webHost from '../../config/web';
 import api from '../../config/web';
 import CreateAlert from './CreateAlert';
+import { PROFILE } from "../../constants/constants";
```
Comment (ashwin319) [inline client/src/Components/Alerts/AlertList.js:8]: in a previous pull request, I had mentioned this `PROFILE` constant should be defined either in a constant called something like `LOCAL_STORAGE_KEYS` or in it's own constants file

Comment (ashwin319) [reply client/src/Components/Alerts/AlertList.js:8]: can you set that as part of this pull request. It's not very clear what PROFILE is supposed to be just by reading the code

File: client/src/Components/Alerts/AlertList.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 
 const AlertList = () => {
   const [list, setList] = useState([]);
   const [loading, setLoading] = useState(true);
   const history = useHistory();
+  const creatorId = JSON.parse(localStorage.getItem(PROFILE)).result._id;
```
Comment (ashwin319) [inline client/src/Components/Alerts/AlertList.js:14]: is this value `localStorage.getItem(PROFILE)` always guaranteed to exist? If not, what should we do to handle scenarios where it doesn't exist?

**Review by ashwin319 — CHANGES_REQUESTED** [review]

File: client/src/Components/Alerts/AlertList.js (hunk 2 above)
Comment (ashwin319) [reply client/src/Components/Alerts/AlertList.js:8]:
```
const LOCAL_STORAGE_KEYS = {
   PROFILE: 'profile'
}
```

Comment (ashwin319) [reply client/src/Components/Alerts/AlertList.js:8]: `import { PROFILE } from "../../constants/localStorageKeys";`

**Review by ashwin319 — APPROVED** [review]

**Review by sergiobtos — APPROVED** [review]

**Pull request merged**